dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/EWeChatSender.java

```diff
@@ -0,0 +1,259 @@
+    }
+
+    private static String textToJsonString(String text) {
+        Map<String, Object> items = new HashMap<>();
+        items.put("msgtype", "markdown");
+        Map<String, Object> textContent = new HashMap<>();
+        byte[] byt = StringUtils.getBytesUtf8(text);
+        String txt = StringUtils.newStringUtf8(byt);
+        textContent.put("content", txt);
+        items.put("markdown", textContent);
+        return JSONUtils.toJsonString(items);
+    }
+
+    private static AlertResult checkSendEWeChatSendMsgResult(String result) {
+        AlertResult alertResult = new AlertResult();
+        alertResult.setStatus("false");
+
+        if (null == result) {
+            alertResult.setMessage("send enterprise wechat msg error");
+            logger.info("send enterprise wechat msg error,enterprise we chat server resp is null");
+            return alertResult;
+        }
+        EWeChatSendMsgResponse sendMsgResponse = JSONUtils.parseObject(result, EWeChatSendMsgResponse.class);
+        if (null == sendMsgResponse) {
+            alertResult.setMessage("send enterprise we chat msg fail");
+            logger.info("send enterprise wechat msg error,resp error");
+            return alertResult;
+        }
+        if (sendMsgResponse.errcode == 0) {
+            alertResult.setStatus("true");
+            alertResult.setMessage("send enterprise wechat msg success");
+            return alertResult;
+        }
+        alertResult.setMessage(String.format("alert send enterprise wechat msg error : %s", sendMsgResponse.getErrmsg()));
+        logger.info("alert send enterprise wechat msg error : {}", sendMsgResponse.getErrmsg());
+        return alertResult;
+    }
+
+    public AlertResult sendEWeChatMsg(String title, String content) {
+        AlertResult alertResult;
+        try {
+            String resp = sendMsg(title, content);
+            return checkSendEWeChatSendMsgResult(resp);
+        } catch (Exception e) {
+            logger.info("send enterprise wechat alert msg  exception : {}", e.getMessage());
+            alertResult = new AlertResult();
+            alertResult.setStatus("false");
+            alertResult.setMessage("send enterprise wechat alert fail.");
+        }
+        return alertResult;
+    }
+
+    private String sendMsg(String title, String content) throws IOException {
+        String data = markdownTableByAlert(title, content);
+        String msg = textToJsonString(data);
+        HttpPost httpPost = constructHttpPost(url, msg, EWeChatAlertConstants.CHARSET);
+
+        CloseableHttpClient httpClient;
+        if (Boolean.TRUE.equals(enableProxy)) {
+            httpClient = getProxyClient(proxy, port, user, passwd);
+            RequestConfig rcf = getProxyConfig(proxy, port);
+            httpPost.setConfig(rcf);
+        } else {
+            httpClient = getDefaultClient();
+        }
+
+        try {
+            CloseableHttpResponse response = httpClient.execute(httpPost);

```

Comment:
Consider using `try-with-resources` to simplify the codes.

```suggestion
        try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
```

and remove the `finally` block

Reply:
Thx.